server/nuxeo-nxr-server/src/main/resources/templates/common-base/client/scripts/taskGC.groovy

```diff
@@ -0,0 +1,33 @@
+import org.apache.logging.log4j.LogManager
+import org.nuxeo.ecm.platform.task.Task
+import org.nuxeo.ecm.core.api.IdRef
+import org.nuxeo.ecm.core.api.ScrollResult
+import org.nuxeo.runtime.transaction.TransactionHelper
+import java.util.HashMap
+
+log = LogManager.getLogger("taskGC.groovy")
+nxql = "SELECT * FROM Document WHERE ecm:mixinType = 'Task'"
+dryRun = true
+session = Context.getCoreSession()
+ScrollResult<String> scrollResult = session.scroll(nxql, 500, 60)
+bucketCount = 0
+documentCount = 0
+
+while (scrollResult.hasResults()) {
+    documentCount += scrollResult.getResults().size()
+    cache = new HashMap<IdRef>()
+    for(id in scrollResult.getResults()) {
+        task = session.getDocument(new IdRef(id)).getAdapter(Task.class)
+        workflowRef = new IdRef(task.getProcessId())
+        if(!cache.computeIfAbsent(workflowRef, session.&exists)) {
+            log.info("removing orphan task " + id)
```

Comment:
You are always logging this, dry run or not, it could be misleading while reading the logs :)

Reply:
Then I will do 2 different logs "would remove..."